src/main/java/io/reactivex/internal/operators/flowable/FlowableMap.java

```diff
@@ -116,15 +116,14 @@ public boolean tryOnNext(T t) {
                 return false;
             }
             
-            if (sourceMode == ASYNC) {
-                actual.onNext(null);
-                return true;
+            if (sourceMode != NONE) {
+                return actual.tryOnNext(null);
             }
             
             U v;
             
             try {
-                v = mapper.apply(t);
+                v = nullCheck(mapper.apply(t), "The mapper function returned a null value.");
```

Comment:
What is `nullCheck`? `Objects.requireNonNull` seems like it meets the same needs. If so let's nuke post-merge.